test/hotspot/jtreg/gc/parallel/TestAlwaysPreTouchBehavior.java

```diff
@@ -0,0 +1,96 @@
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package gc.parallel;
+
+/**
+ * @test TestAlwaysPreTouchBehavior
+ * @summary Tests AlwaysPreTouch Bahavior, pages of java heap should be pretouched with AlwaysPreTouch enabled. This test reads RSS of test process, which should be bigger than heap size(1g) with AlwaysPreTouch enabled.
+ * @requires vm.gc.Serial & os.family == "linux" & os.maxMemory > 2G
+ * @library /test/lib
+ * @run main/othervm -Xmx1g -Xms1g -XX:+UseParallelGC -XX:+AlwaysPreTouch gc.parallel.TestAlwaysPreTouchBehavior
+ */
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.io.IOException;
+import java.util.*;
+import javax.management.*;
+import java.lang.management.*;
+import jdk.test.lib.Utils;
+import jdk.test.lib.Asserts;
+import java.lang.management.*;
+import java.util.stream.*;
+import java.io.*;
+
+public class TestAlwaysPreTouchBehavior {
+    static final long EXCEPTION_VALUE = -1L;
+    public static long getProcessRssInKb() throws IOException {
+        String pid = ManagementFactory.getRuntimeMXBean().getName().split("@")[0];
+        // Read RSS from /proc/$pid/status. Only avaiable on Linux
+        String processStatusFile = "/proc/" + pid + "/status";
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(processStatusFile));
+            StringBuilder stringBuilder = new StringBuilder();
+            String line = null;
+            while ((line = reader.readLine()) != null) {
+                if (line.startsWith("VmRSS:")) {
+                    stringBuilder.append(line);
+                    break;
+                }
+            }
+            stringBuilder.deleteCharAt(stringBuilder.length() - 1);
+            reader.close();
+
+            String content = stringBuilder.toString();
+            return Long.valueOf(content.split("\\s+")[1].trim());
+        } catch (Exception e) {
+           return EXCEPTION_VALUE;
+        }
+
+    }
+    public static void main(String [] args) {
+    long rss = 0;
+    Runtime runtime = Runtime.getRuntime();
+    long committedMemory = (runtime.totalMemory()) / 1024; // in kb
+    try {
+       rss = getProcessRssInKb();
+       System.out.println("RSS = " + rss);
+    } catch (Exception e) {
+       rss = EXCEPTION_VALUE;
+    }
+    if (rss == EXCEPTION_VALUE) {
+        System.out.println("cannot get RSS, just skip");
+        return; // Did not get avaiable RSS, just ignore this test
+    } else if (rss < committedMemory) {
+        System.out.println("RSS = " + rss + " smaller than committed heap memory " + committedMemory);
+    } else {
+        System.out.println("Passed RSS = " + rss + " committed memory " + committedMemory);
+    }
```

Comment:
Why these branches/print here? Could these println be incorporated in the assert msg below?